### ISSUE TITLE
rds wait db-cluster-available

### ISSUE THREAD

**Issue by tomfotherby** [aws/aws-cli#3180]
AWS allows to create a copy-on-write clone of an existing Aurora cluster ([docs](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Aurora.Managing.Clone.html)). When this is done it takes a few minutes before the cluster is available. It would be great if there was a extra `rds wait` command added to [the current list](https://docs.aws.amazon.com/cli/latest/reference/rds/wait/index.html) that works for "clusters" . Example usage:

    rds wait db-cluster-available --db-cluster-identifier <cluster-id>

This would make scripting things easier.. Currently I have a poor workaround such as:

```
status=unknown
while [[ "$status" != "available" ]]; do
  sleep 10
  status=`aws rds describe-db-clusters --db-cluster-identifier $CLUSTERNAME | jq -r '.DBClusters[0].Status'`
done
```

**Comment by dstufft** [comment]
Marking this as an enhancement request, thanks!

**Comment by nhmaha** [comment]
+1

**Comment by jrichard0725** [comment]
+1

**Comment by TBragg** [comment]
+1

**Comment by tbastos** [comment]
+1

**Comment by rfreuler** [comment]
+1

**Comment by stashev** [comment]
It would be nice to have all `wait` command applied to rds Aurora clusters as well, including snapshots

**Comment by josephcarmello** [comment]
+1

**Comment by peterborghard** [comment]
+1

**Comment by IllyaYalovyy** [comment]
+1

**Comment by sebuala** [comment]
More than 1 year since the request and the AWS team couldn't make the enhancement.

**Comment by chandravadans** [comment]
+1

**Comment by shaileshubhe77** [comment]
> AWS allows to create a copy-on-write clone of an existing Aurora cluster ([docs](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Aurora.Managing.Clone.html)). When this is done it takes a few minutes before the cluster is available. It would be great if there was a extra `rds wait` command added to [the current list](https://docs.aws.amazon.com/cli/latest/reference/rds/wait/index.html) that works for "clusters" . Example usage:
> 
> ```
> rds wait db-cluster-available --db-cluster-identifier <cluster-id>
> ```
> 
> This would make scripting things easier. Currently I have a poor workaround:
> 
> ```
> status=unknown
> while [[ "$status" != "available" ]]; do
>   sleep 10
>   status=`aws rds describe-db-clusters --db-cluster-identifier $CLUSTERNAME | jq -r '.DBClusters[0].Status'`
> done
> ```
 correct command is :
> status=unknown
> while [ "$status" != "available" ]; do
>   sleep 10
>   status=$(aws rds describe-db-clusters --db-cluster-identifier $CLUSTERNAME | jq -r '.DBClusters[0].Status' output text)
> done


> AWS allows to create a copy-on-write clone of an existing Aurora cluster ([docs](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Aurora.Managing.Clone.html)). When this is done it takes a few minutes before the cluster is available. It would be great if there was a extra `rds wait` command added to [the current list](https://docs.aws.amazon.com/cli/latest/reference/rds/wait/index.html) that works for "clusters" . Example usage:
> 
> ```
> rds wait db-cluster-available --db-cluster-identifier <cluster-id>
> ```
> 
> This would make scripting things easier. Currently I have a poor workaround:
> 
> ```
> status=unknown
> while [[ "$status" != "available" ]]; do
>   sleep 10
>   status=`aws rds describe-db-clusters --db-cluster-identifier $CLUSTERNAME | jq -r '.DBClusters[0].Status'`
> done
> ```

 correct command is :
> status=unknown
> while [ "$status" != "available" ]; do
>   sleep 10
>   status=$(aws rds describe-db-clusters --db-cluster-identifier $CLUSTERNAME | jq -r '.DBClusters[0].Status' output text)
> done

**Comment by shaileshubhe77** [comment]
Correct command is :
> status=unknown
> while [ "$status" != "available" ]; do
>   sleep 10
>   status=$(aws rds describe-db-clusters --db-cluster-identifier $CLUSTERNAME | jq -r '.DBClusters[0].Status' output text)
> done

**Comment by bpatton00** [comment]
+1

**Comment by jimbocoder** [comment]
+infinity

**Comment by robbhudl** [comment]
+1

**Comment by muriloxk** [comment]
+1

**Comment by reizist** [comment]
+1

**Comment by bbuchalter** [comment]
<img width="275" alt="Screen Shot 2020-06-16 at 11 21 45 AM" src="https://user-images.githubusercontent.com/540762/84806865-94e8f500-afc3-11ea-9e34-a86fce375f5f.png">

**Comment by carojasq** [comment]
+1

**Comment by KapitalEarth** [comment]
I'll pause uploads, then.

Have a good day.

Regards,
R.

**Comment by farman022** [comment]
+1

**Comment by iobites** [comment]
+1

**Comment by paul-m** [comment]
Very Plus One. +1

**Comment by rubengomex** [comment]
+1

**Comment by anthonyruok** [comment]
+1 p-p-p-p-please

**Comment by nchillal** [comment]
+1

**Comment by nic-hima** [comment]
This would be a huge enhancement!

**Comment by tim-finnigan** [comment]
Hi everyone, thanks for your patience regarding this feature request. There is clearly a lot of demand to add this. I raised this internally with the RDS team and currently waiting on a response. In the meantime, I’ll go ahead and transfer this to our shared SDK repository as it is a cross-SDK request.
